### DISCUSSION TITLE
Customizable sort order for issues and PRs

### DISCUSSION THREAD

**Discussion by OndrejSpanel** [community/community#16154] [Projects and Issues]
Currently the default sort order for both Pull Requests and Issues is "Newest". I prefer "Recently updated", as I am most interested in the recently active issues or PRs, but unfortunately GitHub does not remember my preference and I always need to select it again.

I think "Recently updated" is what most other communication platforms usually use. I understand changing the default ordering for an established platform like GitHub is perhaps not an options, but perhaps making the sorting order customizable could be possible?

**Comment by dnut** [comment]
Recently updated should definitely be the default. "Newest" is an insane choice.

**Reply by ADTC** [reply]
According to refined-github/refined-github#80 some people do find "Newest" useful. But yes, I want to default to "Recently updated". This should be a saved user preference at least!

**Reply by ADTC** [reply]
I just realized that the Discussions **_(this page!)_** has "Oldest" as the default sort. I thought "Top" would be the saner choice (bubbling up answers with more votes).

**Reply by dnut** [reply]
"Oldest" is typical for linear discussion threads, like internet forums or github issues, since it makes it easy to follow the timeline of replies. "Top" is more common for branching discussions, like reddit, since a timeline of replies is automatically implied by the branching mechanism regardless of sort order.

On this page, there is only a single level of branching. Nested discussions are linear. This is like stack overflow. I would advocate for the same sorting as stack overflow: "top" for the top level and "oldest" for the nested reply threads.

**Reply by bumblefudge** [reply]
![image](https://github.com/user-attachments/assets/bd832953-baee-4118-8f6c-ebb89c1bbfa6)

**Reply by bumblefudge** [reply]
sort-order default would be great to be able to set at repo or org level, while we're making sensible demands

**Comment by gitgrub** [comment]
Same here, this should be a setting for the user.
I constantly need to switch to "Recently updated", that's cruel.
Installing the browser extension for this feature is useless overkill.

**Reply by OndrejSpanel** [reply]
Can you recommend some browser extension to do this? I would prefer a native solution, but until there is one, I would certainly use an extension.

**Reply by gitgrub** [reply]
Its the one from the post above: https://github.com/orgs/community/discussions/16154#discussioncomment-4396064
I took a look at it at github, but did not install it.

**Comment by gitgrub** [comment]
Maybe the developers problem is where to put that setting into the users profile.
So my proposal would be to use local storage. It is used already on per project basis, so I think this would be a good place to store the last used sort order for each project.

**Comment by Andrej730** [comment]
It's already available in https://github.com/refined-github/refined-github

![image](https://github.com/community/community/assets/9417531/7d3601d6-d342-4231-bda6-44c6104ff1fa)

**Reply by Ryusennin** [reply]
Except Refined GitHub has become a pile of bloat with half of the useful features hidden by default.

**Comment by gitgrub** [comment]
So what does github think about this?

**Comment by aiguofer** [comment]
This and the whitespace option for looking at PRs are my biggest pet peeves with GitHub. All of those kinds of things are configurable in GitLab

**Comment by NilsBaumgartner1994** [comment]
I would suggest to add the option to sort by label order. So displaying issues with labels like "critical" can be seen first.

**Comment by TheGiraffe3** [comment]
Same question as above. What does @github think of this?

**Comment by JansenSmith** [comment]
I would like to be able to set this default, as a repo owner.

**Comment by ramasheshasai** [comment]
Hi there,

That’s a really good point. By default, GitHub always shows “Newest” first for Issues and Pull Requests, and it doesn’t currently remember your preference. Many users (myself included) also find “Recently updated” more useful, since it highlights ongoing discussions and active work.

Changing the global default might be tricky for GitHub, but having a way to customize or save a preferred sort order per user would make things a lot smoother. For now, the only workaround is to bookmark a filtered/sorted view (for example, with sort:updated-desc in the query string), but that’s not as convenient as a built-in preference.

This would make a solid feature request, since remembering sort order on a per-user basis seems like a small but impactful improvement.

**Comment by timdiggins** [comment]
see also https://github.com/orgs/community/discussions/185283
